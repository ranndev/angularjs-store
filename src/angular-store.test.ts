import NgStore from './angularjs-store';

let store: NgStore<typeof state>;
const state = { foo: '', bar: 1, baz: false };

beforeEach(() => {
  store = new NgStore(state);
});

describe('NgStore', () => {
  it('should match the initial state to snapshot', () => {
    expect(store).toMatchSnapshot();
  });

  describe('copy', () => {
    describe('get', () => {
      it('should always return a new copy of state', () => {
        const copies: Array<typeof state> = [];
        for (let i = 0; i < 9; i++) {
          const copy = store.copy();
          expect(copies).not.toContain(copy);
          copies.push(copy);
        }
      });

      it('should return a specific property of state', () => {
        Object.entries(state).forEach(([key, value]) => {
          expect(value).toBe(store.copy(key as keyof typeof state));
        });
      });
    });
  });
});
