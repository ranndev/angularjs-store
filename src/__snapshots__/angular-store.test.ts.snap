// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgStore should match the initial state to snapshot 1`] = `
NgStore {
  "$$hooks": Array [],
  "$$stateHolder": Object {
    "get": [Function],
    "set": [Function],
  },
}
`;
